### DISCUSSION TITLE
NotifyAction with GitKraken and similar

### DISCUSSION THREAD

**Discussion by nostadt** [captainhook-git/captainhook#139] [Q&A]
Hi guys,

in our team we have members who work with the CLI but also with tools as GitKraken. Apparently those who work with tools as GitKraken do not receive those messages by the [NotifyAction](https://github.com/captainhookphp/captainhook/blob/main/src/Hook/Notify/Action/Notify.php) which is a pity. Did anyone here encounter similar issues and probably was able to solve it? Or how do you broadcast important developer news such as "We now use deployer! Check out the README for details."

Looking forward to hear from you.

Alex

PS: Suggestion: Enable [discussions](https://docs.github.com/en/discussions) for this repository.

**Comment by sebastianfeldmann** [comment]
Thanks for the discussions hint :)

I'm not sure how the Cap'n can help here. If the git GUI does not show the git hook output properly the only way to fix this is inside the git GUI.

It would be super nice if GUIs like GitKraken or PHPStorm would provide a way to communicate data from the hook script to the GUI in a better way then displaying some console log that is not shown by default.

**Reply by nostadt** [reply]
Get your point! I am hoping for a workaround or so. It's very individual so not sure if it makes sense for clients to support this OOTB.  What I can imagine is that plugins for a client could do the job. Then clients needs to support plugin feature though... Makes things more unreliable or developer are forced to use specific clients.

**Reply by sebastianfeldmann** [reply]
I opened a feature request for the git-notify syntax for GitKraken and PHPStorm. Let's see what they say.
I would love an API for git hooks to actually do stuff in clients :)

A PHPStorm plugin would actually be absolutely doable 🤔
